### PR TITLE
Fix the deployment target

### PR DESCRIPTION
**Pull request by swapnilgm**
Signed-off-by: Swapnil Mhamane <swapnil.mhamane@sap.com>

**What this PR does / why we need it**:
`make deploy` was failing to deploy druid because wrongly configured kustomization.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: config/default/kustomization.yaml [hunk 1]
```diff
@@ -12,21 +12,20 @@ namePrefix: etcd-druid-
 #commonLabels:
 #  someName: someValue
 
-bases:
-- ../crd
-- ../rbac
-- ../manager
+resources:
```
Comment (georgekuruvillak) [inline config/default/kustomization.yaml:15]: Each entry in this list must resolve to an existing resource definition in YAML.  These are the resource files that kustomize reads, modifies and emits as a YAML string, with resources separated by document markers ("---"). For bases, thats not the case.

Checkout: https://kubectl.docs.kubernetes.io/pages/examples/kustomize.html

File: config/default/kustomization.yaml [hunk 2]
```diff
@@ -12,21 +12,20 @@ namePrefix: etcd-druid-
...
+  - ../crd
+  - ../rbac
+  - ../manager
 
 patches:
-- manager_image_patch.yaml
+  - manager_image_patch.yaml
```
Comment (georgekuruvillak) [inline config/default/kustomization.yaml:21]: Shouldn't this be a list in patches. Why do we need this intendation?

File: config/default/kustomization.yaml [hunk 3]
```diff
@@ -12,21 +12,20 @@ namePrefix: etcd-druid-
...
   # Protect the /metrics endpoint by putting it behind auth.
   # Only one of manager_auth_proxy_patch.yaml and
   # manager_prometheus_metrics_patch.yaml should be enabled.
-#- manager_auth_proxy_patch.yaml
+  #- manager_auth_proxy_patch.yaml
```
Comment (georgekuruvillak) [inline config/default/kustomization.yaml:25]: same as above

File: config/default/manager_image_patch.yaml [hunk 4]
```diff
@@ -2,11 +2,11 @@ apiVersion: apps/v1
...
+  namespace: system
 spec:
   template:
     spec:
       containers:
-      # Change the value of image field below to your controller image URL
-      - image: eu.gcr.io/gardener-project/gardener/etcd-druid:v0.2.0-dev
-        name: druid
+        # Change the value of image field below to your controller image URL
+        - image: eu.gcr.io/gardener-project/gardener/etcd-druid:v0.4.0-dev
```
Comment (georgekuruvillak) [inline config/default/manager_image_patch.yaml:11]: same as above

**Comment by georgekuruvillak** [comment]
Please check the indentations in the yaml files.

File: config/default/kustomization.yaml (hunk 1 above)
Comment (swapnilgm) [reply config/default/kustomization.yaml:15]: Bases is deprecated. https://kubernetes-sigs.github.io/kustomize/api-reference/kustomization/bases/

File: config/default/kustomization.yaml (hunk 2 above)
Comment (swapnilgm) [reply config/default/kustomization.yaml:21]: Thats was default indentation for yaml configured in vscode. And we use same accross repos. I didn't modified it explicitly. And irrespective of that, i do prefer list to be indent, though yaml allows unindented list.

File: config/default/kustomization.yaml (hunk 3 above)
Comment (swapnilgm) [reply config/default/kustomization.yaml:25]: https://github.com/gardener/etcd-druid/pull/60#discussion_r442632871

File: config/default/manager_image_patch.yaml (hunk 4 above)
Comment (swapnilgm) [reply config/default/manager_image_patch.yaml:11]: https://github.com/gardener/etcd-druid/pull/60#discussion_r442632871

**Pull request merged**